debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/ByteDataWrapperHelper.java

```diff
@@ -50,12 +63,24 @@ public static U8 BDWTYPE(ByteDataWrapperPointer ptr) throws CorruptDataException
 		return ptr.dataType();
 	}
 
-	// #define BDWDATA(bdw) (J9SHR_READSRP((bdw)->externalBlockOffset) ? (((U_8*)(bdw)) + J9SHR_READSRP((bdw)->externalBlockOffset)) : (((U_8*)(bdw)) + sizeof(ByteDataWrapper)))	
-	public static U8Pointer BDWDATA(ByteDataWrapperPointer ptr) throws CorruptDataException {
-		if(!ptr.externalBlockOffset().eq(0)) {
-			return U8Pointer.cast(ptr).add(ptr.externalBlockOffset()); 
+	public static U8Pointer getDataFromByteDataWrapper(ByteDataWrapperPointer ptr, U8Pointer[] cacheHeader) throws CorruptDataException {
+		PointerPointer externalBlockOffset = ptr.externalBlockOffsetEA();
+		if (null == cacheHeader) {
+			I32 externalBlockOffsetI32 = I32Pointer.cast(externalBlockOffset.getAddress()).at(0);
+			if (!externalBlockOffsetI32.eq(0)) {
```

Comment:
Please invert this `if` statement to match the one below. Likewise throughout these changes.

Reply:
Fixed